### PR TITLE
add edit and delete customer features

### PR DESCRIPTION
**Pull request by ashleygwilliams**

### CODE REVIEW

**Comment by ashleygwilliams** [comment]
```
Uncaught TypeError: array[(len - 1)].hasOwnProperty is not a function
```

File: app/routes/customer.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ import Ember from 'ember';
 
 export default Ember.Route.extend({
   model: function(customer_id) {
-    return this.store.findRecord('customer', customer_id);
+    var customer = this.store.findRecord('customer', customer_id);
+    return { data: customer, id: customer_id };
```
Comment (ashleygwilliams) [inline app/routes/customer.js]: this is ... https://github.com/emberjs/ember.js/issues/9486

**Comment by ashleygwilliams** [comment]
```
Uncaught Error: Nothing handled the action 'deleteCustomer(model.id)'. If you did handle the action, this error can be caused by returning true from an action handler in a controller, causing the action to bubble.
```

File: app/controllers/customer/delete.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import Ember from 'ember';
+
+export default Ember.Controller.extend({
+  actions: {
+    deleteCustomer: function (customer_id) {
```
Comment (ashleygwilliams) [inline app/controllers/customer/delete.js]: http://guides.emberjs.com/v1.13.0/models/creating-and-deleting-records/

Comment (steveklabnik) [inline app/controllers/customer/delete.js]: so in new, this doens't take an argument, and then does

```
  let customer = this.get('model').save();
```

I wonder if that approach to fetching the model works? and then you don't pass anything when you call `deleteCustomer`

**Comment by ashleygwilliams** [comment]
actually  it works for me. @steveklabnik can u pull this and ensure that delete also works for you?

**Comment by ashleygwilliams** [comment]
@brittanystoroz just gave me a ton of help, will ship this later today

**Comment by steveklabnik** [comment]
:confetti_ball:

File: app/controllers/customers.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import Ember from 'ember';
+
+export default Ember.Controller.extend({
+  actions: {
+    deleteCustomer() {
+      this.get('model')
+        .destroyRecord();
```
Comment (steveklabnik) [inline app/controllers/customers.js]: any reason this isn't on one line like https://github.com/artisan-tattoo/assistant-frontend/pull/30/files#diff-003fa161573a6c036fc4cf470f9b4261R6?

File: app/templates/customer/edit.hbs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      </div>
+      <div class="form-group">
+        <label for="wantInput">Want:</label>
+        {{input value=model.want class="form-control" }}
+      </div>
+      <div class="form-group">
+        <label for="notesInput">Notes:</label>
+        {{textarea value=model.notes class="form-control" rows="3"}}
+      </div>
+      <button class="btn btn-primary" {{action 'editCustomer'}}>Create</button>
```
Comment (steveklabnik) [inline app/templates/customer/edit.hbs]: this says "create" instead of "edit"

**Comment by steveklabnik** [comment]
edit redirects to the first customer for me

**Comment by steveklabnik** [comment]
... and now it didn't

**Comment by steveklabnik** [comment]
I'm also seeing `http://warmachine:4200/customers/undefined` after edit, too.

**Comment by steveklabnik** [comment]
Edits _do_ seem to work, but they don't seem to persist to the database.

**Comment by ashleygwilliams** [comment]
i cant reproduce the edit/redirrct problem, after update it transitions to the detail view of the model

**Comment by steveklabnik** [comment]
My browser was being SUPER weird. This is good now. :+1:

**Pull request merged**